qa/L0_trace/opentelemetry_unittest.py

```diff
@@ -41,27 +40,29 @@
 
 class OpenTelemetryTest(tu.TestResultCollector):
     def setUp(self):
-        while True:
-            with open("trace_collector.log", "rt") as f:
-                data = f.read()
-                if data.count("resource_spans") != EXPECTED_NUM_SPANS:
-                    time.sleep(5)
-                    continue
-                else:
-                    break
-
-        data = data.split("\n")
-        full_spans = [
-            entry.split("POST")[0] for entry in data if "resource_spans" in entry
-        ]
-        self.spans = []
-        self.resource_attributes = []
-        for span in full_spans:
-            span = json.loads(span)
-            self.spans.append(span["resource_spans"][0]["scope_spans"][0]["spans"][0])
-            self.resource_attributes.append(
-                span["resource_spans"][0]["resource"]["attributes"]
+        with open("trace_collector.log", "rt") as f:
+            data = f.read()
+            json_string = re.sub("\n\t{\n\t", "{", data)
+            json_string = re.sub(
+                "resources     : \n\t", "resources     : {\n\t", json_string
+            )
+            json_string = re.sub(
+                "\n  instr-lib     :", "}\n  instr-lib     :", json_string
             )
+            json_string = re.sub(": \n\t", ':"",', json_string)
+            json_string = re.sub(": \n", ':"",', json_string)
+            json_string = re.sub("\n|\n\t", ",", json_string)
+            json_string = re.sub("\t", "", json_string)
+            json_string = re.sub(r"\b([\w.-]+)\b", r'"\1"', json_string)
+            json_string = re.sub('"span" "kind"', '"span kind"', json_string)
+            json_string = re.sub("{,", "{", json_string)
+            json_string = re.sub(",}", "}", json_string)
+            json_string = re.sub("}{", "},{", json_string)
+            json_string = re.sub(
+                '"events"        : {', '"events"        : [{', json_string
+            )
+            json_string = re.sub('},  "links"', '}],  "links"', json_string)
```

Comment:
Can you explain the goal of this part? I find this hard to follow and figure there may be an easier way just working with dictionaries and then reserializing to JSON or something.

Reply:
The problem is that OStream exporter does not export spans in a json format (i.e. simple `json.loads(json.dumps(data))` won't work), so extracted spans need to be transformed to json. I haven't finished working on it yet, this was put in here for testing on our CI, so it may be simplified later